src/core/tasks.py

```diff
@@ -157,6 +159,48 @@ def as_dict(self):
             endpoint["response_path"] = self.response_path
         return endpoint
 
+class Wait(JsonRpc):
+    __slots__ = [
+    "timeout",
+    "conditions",
+    ]
+
+    def __init__(
+        self,
+        name,
+        preconditions=None,
+        url=None,
+        method=None,
+        payload_paths=None,
+        payload=None,
+        response_path=None,
+        timeout=None,
+        conditions=None,
+    ):
+        super().__init__(
+            name=name, 
+            preconditions=preconditions, 
+            url=url, 
+            method=method, 
+            payload_paths=payload_paths, 
+            payload=payload,
+            response_path=response_path,
+            task_type="wait")
+        
+        self.timeout = timeout
+        self.conditions = conditions
+
+    def get_conditions(self):
```

Comment:
you still need `get_validators` with a super call and this function in it.

Reply:
Yeh, didn't see it at first, you're right there :)

Reply:
@nocturnalastro Does the below function resolve this?

`    def get_validators(self):
        yield from super().get_validators()
        yield from self.conditions`

Reply:
Yes that shoud be fine

btw try to use tripple back ticks ``` when putting code in comments stops githubs formatting from f-ing it up